tests/agents/test_orchestrator.py

```
@@ -0,0 +1,366 @@
+    @pytest.mark.asyncio
+    async def test_handle_agent_failure(self, orchestrator):
+        """Should handle agent execution failures gracefully."""
+        failing_agent = MagicMock()
+        failing_agent.id = "failing-agent"
+        failing_agent.execute = AsyncMock(side_effect=Exception("Agent crashed"))
+
+        if hasattr(orchestrator, 'execute_task'):
+            task = {"id": "task-fail", "agent_id": failing_agent.id}
+
+            with patch.object(orchestrator, 'get_agent', return_value=failing_agent):
+                with pytest.raises(Exception) or result:
```

Flake8 (7.3.0)

[error] undefined name 'result'

(F821)